tests/unit/test_config.py

```diff
@@ -203,6 +203,26 @@ def test_add_all_interfaces_value(self):
             HostAndPort("0.0.0.0", 42),
         ]
 
+    def test_add_all_interfaces_value_ipv6(self):
+        ports = config.UniqueHostAndPortList()
+        ports.append(HostAndPort("::", 42))
+        ports.append(HostAndPort("::1", 42))
+
+        assert ports == [
+            HostAndPort("::", 42),
+        ]
+
+    def test_add_all_interfaces_value_mixed_ipv6_wins(self):
+        ports = config.UniqueHostAndPortList()
+        ports.append(HostAndPort("0.0.0.0", 42))
+        ports.append(HostAndPort("::", 42))
+        ports.append(HostAndPort("127.0.0.1", 42))
+        ports.append(HostAndPort("::1", 42))
```

Comment:
q: is it valid to listen on both `127.0.0.1` and `::1`?

Reply:
It's nonsensical to bind to both, because `::1` includes `127.0.0.1`, and Twisted won't let you open listeners for both. This is why I made the logic remove `127.0.0.1` if `::1` has also been added.

Reply:
I don't think this is covered by the tests. I can see tests that ensure that if `::` is added then this "trumps" everything, but I don't see a test covering adding `::` and `127.0.0.1`.

I have written this test and it fails:


```python
    def test_loopback_ipv6_only(self):
        ports = config.UniqueHostAndPortList()
        ports.append(HostAndPort("::1", 4566))
        ports.append(HostAndPort("127.0.0.1", 4566))

        assert ports == [
            HostAndPort("::", 4566),
        ]
```
Would it be possible to fix this, then I am happy with this PR